### ISSUE TITLE
Confusing mixture of single-job and multi-job Pipeline schema

### ISSUE THREAD

**Issue by escalonn** [MicrosoftDocs/vsts-docs#1711]
The presentation of the Pipeline schema is made needlessly confusing by failing to distinguish between single-job and multi-job syntax. It looks at first like variables with pipeline scope are possible, but (at least a few days ago) `variables:` can only be specified with the single-job format. Yet `jobs:` is also shown right after `variables:`.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7458d5b2-1bb7-7d87-7ce6-2b0c430b0155
* Version Independent ID: edf5bbd3-a3f0-5426-e985-31290f8afd05
* Content: [YAML schema - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=vsts)
* Content Source: [docs/pipelines/yaml-schema.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/yaml-schema.md)
* Product: **devops**
* GitHub Login: @erickson-doug
* Microsoft Alias: **macoope**

**Comment by escalonn** [comment]
You're right, looks like I was mistaken.

One source for my confusion is the linked [variables page](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables) says that variables are scoped to the job and suggests that only output variables can be multi-job.

I think the schema page would be improved by being more formally precise like language grammars generally are, and by better distinguishing the schema definition code blocks from the example code blocks.

The `pool:` section of the schema definition is unlike the others in that the code block doesn't include the top-level `pool:` bit. And the higher-level "job" section of the schema doesn't reference "pool" - instead, it references "server", which has a much vaguer definition.

Maybe I've got the wrong expectations for this whole system. For example it's strange to me that you don't version the YAML schema as a whole - isn't that a decision usually regretted later? I was impressed when I first learned Tasks were versioned, and the same reasoning would apply here. But you were just asking about this page, so enough about that.

**Comment by vtbassmatt** [comment]
I'll fix the `pool:` inconsistency, thank you for pointing it out. And thanks for catching that error on the variables page; we changed behavior recently and I missed it.

Versioning the YAML schema is something we will look at when there are breaking changes. At the moment, schema additions have only been additive. I removed the documentation about the deprecated parts as we want to steer people towards the new syntax.

This doc was a move towards formality while remaining accessible and useful to people. I'll see if there are ways I can tighten it up even further. Part of the challenge is that we're somewhat liberal in what we accept, which is hard to document in a readable way.
